source/client/factories_impl.cc

```diff
@@ -118,16 +118,24 @@ void RequestSourceFactoryImpl::setRequestHeader(Envoy::Http::HeaderMap& header,
 }
 
 RequestSourcePtr RequestSourceFactoryImpl::create() const {
-  // Note: we assume a valid uri.
-  // Also, we can't resolve, but we do not need that.
-  UriImpl uri(options_.uri());
   Envoy::Http::HeaderMapPtr header = std::make_unique<Envoy::Http::HeaderMapImpl>();
+  if (options_.uri().has_value()) {
+    // Note: we assume a valid uri.
+    // Also, we can't resolve, but we do not need that.
+    UriImpl uri(options_.uri().value());
+    header->setPath(uri.path());
+    header->setHost(uri.hostAndPort());
+    header->setScheme(uri.scheme() == "https" ? Envoy::Http::Headers::get().SchemeValues.Https
+                                              : Envoy::Http::Headers::get().SchemeValues.Http);
+  } else {
+    header->setPath(options_.multiTargetPath());
+    header->setHost("dummy");
```

Comment:
Can we improve the string literal to make this a bit more self-documenting? E.g. "host is not used with multi target" or similar.

Reply:
Done